File: src/main/java/repository/IReservationRepository.java
package repository;

import model.Reservation;
import java.util.List;

public interface IReservationRepository {
    void save(Reservation reservation);
    Reservation findById(int id);
    List<Reservation> findAll();
    List<Reservation> findByTrip(int tripId);
    List<Reservation> findByAgency(int agencyId);
    void delete(int id);
    void update(Reservation reservation);
    void resequenceIds();
}

File: TurismUI/TurismClient/Repositories/IReservationRepository.cs
using System.Collections.Generic;
using TurismClient.Models;

namespace TurismClient.Repositories;

public interface IReservationRepository
{
    void Save(Reservation reservation);
    Reservation? FindById(int id);
    IEnumerable<Reservation> FindAll();
    IEnumerable<Reservation> FindByTrip(int tripId);
}